vxquery-core/src/main/java/org/apache/vxquery/datamodel/accessors/jsonItem/ObjectPointable.java

```diff
@@ -0,0 +1,155 @@
+            return VoidPointable.TYPE_TRAITS;
+        }
+
+        @Override
+        public IPointable createPointable() {
+            return new ObjectPointable();
+        }
+    };
+    private static final int ENTRY_COUNT_SIZE = IntegerPointable.TYPE_TRAITS.getFixedLength();
+    private static final int SLOT_SIZE = 4;
+    private final ArrayBackedValueStorage abvs = new ArrayBackedValueStorage();
+    private final GrowableIntArray slots = new GrowableIntArray();
+    private final ArrayBackedValueStorage dataArea = new ArrayBackedValueStorage();
+
+    private static int getSlotValue(byte[] bytes, int start, int idx) {
+        return IntegerPointable.getInteger(bytes, getSlotArrayOffset(start) + idx * SLOT_SIZE);
+    }
+
+    private static int getEntryCount(byte[] bytes, int start) {
+        return IntegerPointable.getInteger(bytes, start);
+    }
+
+    private static int getKeyLength(byte[] b, int s) {
+        return UTF8StringPointable.getUTFLength(b, s) + 2;
+    }
+
+    private static int getSlotArrayOffset(int start) {
+        return start + ENTRY_COUNT_SIZE;
+    }
+
+    private static int getDataAreaOffset(byte[] bytes, int start) {
+        return getSlotArrayOffset(start) + getEntryCount(bytes, start) * SLOT_SIZE;
+    }
+
+    public void getKeys(IPointable result) throws SystemException {
+        try {
+            abvs.reset();
+            slots.clear();
+            dataArea.reset();
+            int dataAreaOffset = getDataAreaOffset(bytes, start);
+            int entryCount = getEntryCount();
+            int s;
+            for (int i = 0; i < entryCount; i++) {
+                s = dataAreaOffset + getRelativeEntryStartOffset(i);
+                dataArea.getDataOutput().write(ValueTag.XS_STRING_TAG);
+                dataArea.getDataOutput().write(bytes, s, getKeyLength(bytes, s));
+                slots.append(dataArea.getLength());
+            }
+            finishSequenceBuild();
+            result.set(abvs);
+        } catch (IOException e) {
+            throw new SystemException(ErrorCode.SYSE0001);
+        }
+    }
+
+    private void finishSequenceBuild() throws IOException {
+        DataOutput out = abvs.getDataOutput();
+        if (slots.getSize() != 1) {
+            out.write(ValueTag.SEQUENCE_TAG);
+            int size = slots.getSize();
+            out.writeInt(size);
+            if (size > 0) {
+                int[] slotArray = slots.getArray();
+                for (int i = 0; i < size; ++i) {
+                    out.writeInt(slotArray[i]);
+                }
+                out.write(dataArea.getByteArray(), dataArea.getStartOffset(), dataArea.getLength());
+            }
+        } else {
+            out.write(dataArea.getByteArray(), dataArea.getStartOffset(), dataArea.getLength());
+        }
+    }
+
+    public void getValue(TaggedValuePointable key, TaggedValuePointable result) throws SystemException {
+        int dataAreaOffset = getDataAreaOffset(bytes, start);
```

Comment:
You can change this function to only accept UTF8StringPointables for the key argument. The caller should handle the tagged value.